app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -126,18 +127,45 @@ const Lease = ({ form, values, store }) => {
     ...map(confirmedPreferences, pluck('id', 'preference_name'))
   ])
 
-  const getVisited = (fieldName) => form.getFieldState(fieldName)?.visited
+  const availableUnits = units.filter(
+    (unit) => !unit.application_id || unit.application_id === application.id
+  )
 
-  const openDeleteLeaseConfirmation = () => setShowDeleteConfirmation(true)
-  const closeDeleteLeaseConfirmation = () => setShowDeleteConfirmation(false)
+  const availableUnitsOptions = formUtils.toOptions(
+    map(availableUnits, pluck('id', 'unit_number', 'priority_type'))
+  )
 
-  const handleDeleteLeaseConfirmed = () => {
-    setShowDeleteConfirmation(false)
-    handleDeleteLease()
-  }
+  const selectedUnit = form.getState().values.lease?.unit
 
   const areNoUnitsAvailable = !availableUnitsOptions.length
 
+  const prefUsedIs = (pref) => {
+    const prefUsed = confirmedPreferences.find(
+      (pref) => pref.id === form.getState().values.lease?.preference_used
+    )
+    return prefUsed?.preference_name.includes(pref)
+  }
+
+  const unitIsUsedWithPrefElsewhere = (unit, pref) =>
+    unit.application_id &&
+    unit.application_id !== application.id &&
+    unit.preference_used_name?.includes(pref)
+
+  const remainingSetAsidesForPref = (pref) =>
+    totalSetAsidesForPref(listing, pref) -
+    units.filter(
```

Comment:
I think this could be more readable if this value was pulled out into a const named like `numUnitsUsingPref`. i was just a little confused when I first looked at this